### PR TITLE
Add Username Update and Profile Picture Upload Features

### PR DESCRIPTION
**Pull request by halimyu**
### Username Update:
- Implemented functionality to allow users to update their username from the **Settings page**.
- Added a form for editing the username with validation to ensure proper input.
- Displayed success or error messages based on the update outcome.
- Integrated the feature with:
  - **RSpec tests**: Verified success and failure scenarios for updating the username.
  - **Cucumber tests**: Tested the username update workflow, including form submission and validation handling.

### Profile Picture Upload:
- Added functionality to upload and change the user's profile picture via a modal on the **Settings page**.
- Used **Active Storage** for handling file attachments:
  - Configured Active Storage for the application.
  - Added support for storing and displaying profile images, with a fallback to a default image if none is uploaded.
- Ensured user experience with:
  - A visually styled circular profile picture display.
  - A modal interface for uploading new images, complete with validation.
- Integrated the feature with:
  - **RSpec tests**: Verified valid image uploads, missing image handling, and error scenarios.
  - **Cucumber tests**: Tested the full workflow of opening the modal, uploading an image, and verifying the update.

## Testing
- Both **RSpec** and **Cucumber** tests were created and successfully passed for both features:
  - Username update.
  - Profile picture upload.
- Verified functionality in the UI:
  - Ensured forms display correctly and are responsive.
  - Confirmed that flash messages are shown appropriately for success and error cases.

---

## Notes
These changes enhance the **Settings page** by providing a more personalized and user-friendly experience. Users can now update their profile information seamlessly. Further enhancements can focus on additional user preferences or real-time updates for these features.

## Preview

image
image
image
image
image

### CODE REVIEW

File: features/change_email.feature
Comment (bjohnson66) [inline features/change_email.feature:1]: Why did you remove this (*cries)


It still works so its probably fine

**Review by bjohnson66 — APPROVED** [review]
Seems to work well. GREAT WORK!

**Review by zhasa18 — APPROVED** [review]
Love to see it!
